CIScripts/Tests/VRouterAgentTests.ps1

```diff
@@ -70,17 +72,35 @@ function Test-VRouterAgentIntegration {
     function Assert-AgentIsRunning {
         Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session)
 
-        if (!(Test-IsVRouterAgentEnabled -Session $Session)) {
-            throw "vRouter Agent is not running. EXPECTED: vRouter Agent is running"
+        $MaxNumberOfChecks = 10;
+        $SleepTimeBetweenChecks = 10;
+
+        for ($RetryNum = $MaxNumberOfChecks; $RetryNum -gt 0; $RetryNum--) {
+            if (Test-IsVRouterAgentEnabled -Session $Session) {
+                return
+            }
+
+            Start-Sleep -s $SleepTimeBetweenChecks
         }
+
+        throw "vRouter Agent is not running. EXPECTED: vRouter Agent is running"
     }
 
     function Assert-AgentIsNotRunning {
         Param ([Parameter(Mandatory = $true)] [System.Management.Automation.Runspaces.PSSession] $Session)
 
-        if (Test-IsVRouterAgentEnabled -Session $Session) {
-            throw "vRouter Agent is running. EXPECTED: vRouter Agent is not running"
+        $MaxNumberOfChecks = 10;
+        $SleepTimeBetweenChecks = 10;
```

Comment:
Duplicate constants in `Assert-AgentIsRunning`. Should be moved somewhere else IMO.